addons/sale_coupon/models/coupon_program.py

```diff
@@ -12,19 +11,19 @@ class CouponProgram(models.Model):
 
     # The api.depends is handled in `def modified` of `sale_coupon/models/sale_order.py`
     def _compute_order_count(self):
-        program_orders = self.env["sale.order.line"].read_group(
+        program_orders = self.env['sale.order.line'].read_group(
             domain=[
-                ("product_id", "in", self.mapped("discount_line_product_id").ids),
+                ('product_id', 'in', self.discount_line_product_id.ids),
             ],
-            fields=["order_id"],
-            groupby=["product_id", "order_id"],
-            lazy=False,
+            fields=['order_id:count_distinct'],
```

Comment:
I tried that and it just wouldn't actually count the distinct orders. I must have been doing sth wrong :shrug:

Reply:
Well, unless I missed something, it seemed to be working fine on my side :D